scripts/exrrfs_mpassit.sh

```diff
@@ -37,20 +26,78 @@ elif [[ "${MESH_NAME}" == "conus3km" ]]; then
   dx=3000.0
   ref_lat=38.5
 fi
-sed -e "s/@timestr@/${timestr}/" -e "s/@nx@/${nx}/" -e "s/@ny@/${ny}/" -e "s/@dx@/${dx}/" \
-    -e "s/@ref_lat@/${ref_lat}/" ${PARMrrfs}/namelist.mpassit > namelist.mpassit
+#
+#  find cycle time
+#
+YYYYMMDDHH=${CDATE}
+YYYYMMDD=${CDATE:0:8}
+HH=${CDATE:8:2}
+#
+#  find the localtion of the history files
+#
+if [[ -z "${ENS_INDEX}" ]]; then
+  ensindexstr=""
+else
+  ensindexstr="/mem${ENS_INDEX}"
+fi
+history_dir=${UMBRELLA_DATA}${ensindexstr}/${RUN}_fcst_${cyc}
+#
+# find forecst length for this cycle
+#
+fcst_length=${FCST_LENGTH:-1}
+fcst_len_hrs_cycles=${FCST_LEN_HRS_CYCLES:-"01 01"}
+fcst_len_hrs_thiscyc=$(${USHrrfs}/find_fcst_length.sh "${fcst_len_hrs_cycles}" "${cyc}" "${fcst_length}")
+echo "forecast length for this cycle is ${fcst_len_hrs_thiscyc}"
+#
+# loop through forecast history files for this group
+#
+fhr_string=$( seq 0 $((10#${HISTORY_INTERVAL})) $((10#${fcst_len_hrs_thiscyc} )) )
+fhr_all=(${fhr_string})
+num_fhrs=${#fhr_all[@]}
+group_total_num=$((10#${GROUP_TOTAL_NUM}))
+group_index=$((10#${GROUP_INDEX}))
+
+for (( ii=0; ii<${num_fhrs}; ii=ii+${group_total_num} )); do
+    i=$(( ii + ${group_index} - 1 ))
+    if (( $i >= ${num_fhrs} )); then
+      break
+    fi
+# get forecast hour and string
```

Comment:
Can we indent all the comment lines to align with their corresponding codes?

Reply:
Changed for exrrfs_mpassit.sh, exrrfs_save_fcst.sh, exrrfs_upp.sh